examples/mugen/retrieval/eval.py

```diff
@@ -4,114 +4,50 @@
-        "debug": {"action": "store_true"},
-    },
-    "videoclip": {
-        "text_pretrained": {"action": "store_true"},
-        "text_trainable": {"action": "store_true"},
-        "text_model_name": {"type": str, "default": "distilbert-base-uncased"},
-        "text_model_config": {"type": json.loads, "default": None},
-        "text_padding_value": {"type": int, "default": 0},
-        "video_pretrained": {"action": "store_true"},
-        "video_trainable": {"action": "store_true"},
-        "video_pretrain_path": {"type": str, "default": PRETRAINED_S3D_KINETICS400_URL},
-        "proj_out_dim": {"type": int, "default": 256},
-        "proj_dropout": {"type": float, "default": 0.1},
-    },
-    "evaluation": {
-        "accelerator": {"type": str, "default": "auto"},
-        "devices": {"type": int, "default": 1},
-        "checkpoint_path": {
-            "type": str,
-            "default": "lightning_videoclip_mugen_ckpt.pt",
-        },
-    },
-}
-
-
-def parse_all_args():
-    parser = argparse.ArgumentParser()
-    for arguments_dict in arg_structure.values():
-        for arg_name, arg_options in arguments_dict.items():
-            parser.add_argument("--" + arg_name, **arg_options)
-    args = parser.parse_args()
-    return args
-
-
-def get_args_from_group(args, group_name):
-    """Utility for grouping arguments based on the structure in ``arg_structure``"""
-    return argparse.Namespace(
-        **{
-            arg_name: getattr(args, arg_name)
-            for arg_name in arg_structure[group_name].keys()
-        }
-    )
+def get_yaml_config():
+    cli_conf = OmegaConf.from_cli()
+    if "config" not in cli_conf:
+        raise ValueError(
+            "Please pass 'config' to specify configuration yaml file for running VideoCLIP evaluation"
+        )
+    yaml_conf = OmegaConf.load(cli_conf.config)
+    return OmegaConf.to_container(yaml_conf)
 
 
 def evaluate():
-    args = parse_all_args()
-    datamodule_args = get_args_from_group(args, "datamodule")
-    lightningmodule_args = get_args_from_group(args, "lightningmodule")
-    dataset_args = get_args_from_group(args, "dataset")
-    videoclip_args = get_args_from_group(args, "videoclip")
-    evaluation_args = get_args_from_group(args, "evaluation")
-
-    dataset_args = MUGENDatasetArgs(get_audio=False, **vars(dataset_args))
+    args = get_yaml_config()
+    datamodule_args = args["datamodule_args"]
+    lightningmodule_args = args["lightningmodule_args"]
+    dataset_args = args["dataset_args"]
+    videoclip_args = args["videoclip_args"]
+    evaluation_args = args["evaluation_args"]
+
+    dataset_args = MUGENDatasetArgs(get_audio=False, **dataset_args)
     datamodule = MUGENDataModule(
         dataset_args,
-        text_transform=BertTextTransform(),
-        video_transform=VideoTransform(),
-        **vars(datamodule_args),
+        text_transform=BertTextTransform(**datamodule_args["bert_text_transform"]),
```

Comment:
Why do we need items with `{}` value in the yaml file? Looks like they are bloating your constructor APIs here.